jcl/src/java.management/share/classes/com/ibm/java/lang/management/internal/ThreadMXBeanImpl.java

```diff
@@ -756,4 +779,44 @@ public ObjectName getObjectName() {
 		return objectName;
 	}
 
+	private static MethodHandle getThreadInfoConstructorHandle() {
+		return AccessController.doPrivileged(new PrivilegedAction<MethodHandle>() {
+			@Override
+			public MethodHandle run() {
+				MethodHandle myHandle = null;
+				try {
+					Constructor<ThreadInfo> threadInfoConstructor = ThreadInfo.class
+							.getDeclaredConstructor(ThreadInfoBase.class);
+					threadInfoConstructor.setAccessible(true);
+					myHandle = MethodHandles.lookup().unreflectConstructor(threadInfoConstructor);
+				} catch (NoSuchMethodException | SecurityException | IllegalAccessException e) {
+					/*[MSG "K0614", "Unexpected exception accessing ThreadInfo constructor."]*/
+					throw new RuntimeException(com.ibm.oti.util.Msg.getString("K0614"), e); //$NON-NLS-1$
+				}
+				return myHandle;
+			}
+		});
+	}
+	
+
+	private static ThreadInfo makeThreadInfo(ThreadInfoBase base) {
+		ThreadInfo newThreadInfo = null;
+		if ((null != base) && (null != threadInfoConstructorHandle)) {
+			newThreadInfo = AccessController.doPrivileged(new PrivilegedAction<ThreadInfo>() {
+
+				@Override
+				public ThreadInfo run() {
+					ThreadInfo result = null;
+					try {
+						result = (ThreadInfo) threadInfoConstructorHandle.invoke(base);
+					} catch (Throwable e) {
```

Comment:
This should not be catching `Throwable`; instead the body of `run()` should be:
```
    Throwable trouble;

    try {
        return threadInfoConstructorHandle.invoke(base);
    } catch (IllegalAccessException e) {
        trouble = e;
    } catch (InvocationTargetException e) {
        trouble = e.getCause();

        if (trouble instanceof RuntimeException) {
            throw (RuntimeException) trouble;
        }

        if (trouble instanceof Error) {
            throw (Error) trouble;
        }
    }

    /*[MSG "K0614", "Unexpected exception accessing ThreadInfo constructor."]*/
    throw new RuntimeException(com.ibm.oti.util.Msg.getString("K0614"), trouble); //$NON-NLS-1$
```

Reply:
Thanks.  Code updated accordingly.  Also created separate message for the loop and invocation code.